src/react-native/01-intro-to-react-native/intro-to-react-native.md

```diff
@@ -2,47 +2,76 @@
 
-TODO
+**State encapsulation:** Components in React can hold and manage their own state. This encapsulation of state in components makes for a more predictable and easier-to-manage state management, especially in large-scale applications.
 
-### Exercise 1
+### JSX for templating
 
-TODO
+React’s declarative nature means you define UIs as they should appear based on the current state, using JSX. This syntax allows you to write your component’s structure in a way that’s similar to HTML, differing from traditional imperative methods of UI construction.
 
-### Solution 1
+Here’s why JSX is highly regarded as a templating language:
 
-If you’ve implemented the solution correctly, the tests will pass when you run `npm run test`!
+**Familiarity with HTML:** JSX closely resembles HTML, making it intuitive for anyone familiar with web development. This familiarity allows us to quickly understand and write UI code, significantly reducing the learning curve for new React developers.
 
-<details>
-<summary>Click to see the solution</summary>
+**Enhanced readability:** JSX promotes cleaner and more readable code. It allows you to clearly see the layout of your UI in a way that feels like writing HTML, making it easier to visualize the component’s structure and behavior.
 
-TODO
+### UI as a function of data and state
 
-</details>
+React maps the JSX that we write to a Virtual DOM, which is an in-memory representation of the real browser DOM. This apprach has numerous benefits in building large applications:
+
+**Predictable state-to-UI mapping:** React excels at ensuring that the UI is always a direct reflection of the application’s state. This approach means that every time there’s a change in state, React updates the Virtual DOM to reflect this change. As a developer, this gives you a predictable and straightforward way to manage UI changes, as the state drives the UI rendering.
+
+**Focus on state, not the DOM:** The Virtual DOM lets us concentrate on managing the state of the application. The UI updates automatically to mirror the current state, thanks to React’s reactive model. This significantly simplifies the development process, as you’re not manually handling the DOM updates, leading to more maintainable and readable code.
+
+**Efficient update mechanism:** React’s rendering approach leads to efficient update mechanisms in dynamic UIs. By batching these updates and only applying the necessary changes to the actual DOM, React ensures that the UI is always in sync with the state with minimal performance overhead.
+
+**Minimized reflows and repaints:** Direct DOM manipulations can be costly, especially in complex UIs. React’s Virtual DOM minimizes the direct interactions with the actual DOM, reducing costly reflows and repaints. This results in a smoother performance, particularly in applications where the UI changes frequently based on user interactions and state changes.
```

Comment:
I think a lot of this content belongs on the React DOM page and not here. Some of it is just not applicable to React Native.

For React Native, I think the “what” should focus on:

- React as the UI library.
- JavaScript as the language.
- Cross-platform Android and iOS apps.
- Technically, a JS runtime that bridges to native Android/iOS components/views.

The “why” should focus on:

- Build Android and iOS apps with one codebase, not multiple.
- Teams focus on one language (JS).
- JS can be reused across mobile and web.
- RN renders as native components, not a webview, which can be more performant.
- RN is more mature than the native declarative options (Jetpack Compose and SwiftUI).